src/ddsts/include/dds/ddsts/typetree.h

```diff
@@ -429,6 +433,87 @@ ddsts_create_declaration(ddsts_identifier_t name, ddsts_type_t *decl_type, ddsts
 DDS_EXPORT dds_return_t
 ddsts_declaration_set_type(ddsts_type_t *declaration, ddsts_type_t *type);
 
+/**
+ * @brief Creates a ddsts_forward_t struct for a union forward declaration.
+ *
+ * @param[in]   name    A non-NULL pointer to a string. If the function
+ *                      returns success, the string should be considered as
+ *                      owned by the created ddsts_forward_t struct.
+ * @param[out]  result  Pointer to the created ddsts_forward_t struct.
+ *
+ * @returns A dds_return_t indicating success or failure.
+ */
+DDS_EXPORT dds_return_t
+ddsts_create_union_forward_dcl(ddsts_identifier_t name, ddsts_type_t **result);
+
+/* Union declaration (union_def)
+ */
+typedef struct {
+  ddsts_typespec_t type;
+  ddsts_flags_t switch_type;
```

Comment:
At the moment, all scoped names are resolved during the parsing, and not stored as such in the Type Tree. So far, this seems to work in the remainder of the code. If you know of a case where this is not going to work, I am happy to hear about it.

Reply:
This might become an issue when dealing with a discriminator based on an enum. Of course you could choose to store just the values of the enum, but it might make sense to actually store the labels. For example, if you want to generate the type descriptor for a union, you might want to refer to the enum labels instead of to the values they represent.